### ISSUE TITLE
[camelCase] broken with flow-type

### ISSUE THREAD

**Issue by SBoudrias** [eslint/eslint#11529]
**Tell us about your environment**

* **ESLint Version:** `5.15.3` (works with `5.15.0`)
* **Node Version:** `v10.15.0`
* **npm Version:** `6.4.1`

**What parser (default, Babel-ESLint, etc.) are you using?** `babel-eslint`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
    parser: 'babel-eslint',
    extends: [
        'airbnb',
        'prettier',
        'prettier/flowtype',
        'prettier/react',
        'plugin:jest/recommended',
    ],
    env: {
        jest: true,
        node: true,
        'cypress/globals': true,
    },
    plugins: ['flowtype', 'prettier', 'gettext', 'jest', 'cypress', 'react-hooks'],
    settings: {
        'import/resolver': {
            'babel-module': {},
            alias: [['sentry', '@sentry/browser']],
        },
    },
    rules: {
        'jest/valid-expect-in-promise': 'error',
        'jest/no-test-prefixes': 'error',
        'jest/no-large-snapshots': 'warn',
        'jest/consistent-test-it': ['error', { fn: 'it' }],
        'jest/prefer-to-have-length': 'error',
        'jest/prefer-to-be-null': 'error',
        'jest/prefer-to-be-undefined': 'error',
        'gettext/no-variable-string': 'error',
        'gettext/required-positional-markers-for-multiple-variables': 'error',
        'array-callback-return': 'off',
        indent: 'off',
        'no-return-assign': 'off',
        'no-underscore-dangle:': 'off',
        'no-unused-expressions': ['error', { allowTaggedTemplates: true }],
        'no-underscore-dangle': 'off',
        'lines-between-class-members': 'off',
        'flowtype/use-flow-type': 'error',
        'flowtype/define-flow-type': 'error',
        'flowtype/require-valid-file-annotation': ['error', 'always'],
        'prefer-rest-params': 'off',
        'prefer-destructuring': 'off',
        'react/default-props-match-prop-types': [
            'error',
            { allowRequiredDefaults: true },
        ],
        'react/jsx-filename-extension': ['error', { extensions: ['.js'] }],
        'react/jsx-indent': 'off',
        'react/jsx-indent-props': 'off',
        'react/destructuring-assignment': 'off',
        'react/no-array-index-key': 'off',
        'react/no-multi-comp': 'off',
        'react/no-unused-prop-types': 'off',
        'react/require-default-props': 'off',
        'react/sort-comp': 'off',
        'react-hooks/rules-of-hooks': 'error',
        'react-hooks/exhaustive-deps': 'warn',
        'jsx-a11y/anchor-has-content': 'off',
        'jsx-a11y/href-no-hash': 'off',
        'jsx-a11y/label-has-associated-control': ['error', { depth: 2 }],
        'jsx-a11y/label-has-for': 'off',
        'jsx-a11y/no-static-element-interactions': 'off',
        'jsx-a11y/html-has-lang': 'off',
        'import/no-cycle': 'off',
        'import/no-extraneous-dependencies': [
            'error',
            {
                devDependencies: [
                    '**/test.js',
                    '**/*.test.js',
                    '**/stories.js',
                    'jest.setup.js',
                    'next.config.js',
                    'postcss.config.js',
                    'server.js',
                    'bin/**',
                    'docker/cypress/**',
                    '**/build-cities.js',
                ],
            },
        ],
        'import/prefer-default-export': 'off',
        'prettier/prettier': 'error',
        'import/no-absolute-path': 'off',
    },
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
function getInitialProps({
    pathname,
    asPath,
    query,
    req,
    res,
}: {
    pathname: string,
    asPath: string,
    query: { sdk_slug: string, queries?: string },
    req: ?http$IncomingMessage<*>,
    res: ?http$ServerResponse,
}) {
    // ...
}
```

**What did you expect to happen?**

`camelCase` shouldn't alert on flow-type keys if the configuration is set as `camelcase: ['error', { properties: 'never' }]`

**What actually happened? Please include the actual, raw output from ESLint.**

Here the `camelCase` rule errored on the `sdk_slug` key inside the flow type definition.

**Comment by g-plane** [comment]
There is similar issue with TypeScript.
